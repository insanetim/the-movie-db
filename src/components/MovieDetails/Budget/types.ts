import { IMovieDetails } from 'src/interfaces/movie.interface'

export type BudgetProps = {
  budget: IMovieDetails['budget']
}
