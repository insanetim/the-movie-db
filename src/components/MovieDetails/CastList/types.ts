import { IMovieCredit } from 'src/interfaces/movie.interface'

export type CastListProps = {
  cast: IMovieCredit[]
  showTotal?: boolean
}
