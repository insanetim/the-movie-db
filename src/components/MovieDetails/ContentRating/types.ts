export type ContentRatingProps = {
  contentRating: string
}
