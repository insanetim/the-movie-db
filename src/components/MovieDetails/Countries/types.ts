import { IMovieDetails } from 'src/interfaces/movie.interface'

export type CountriesProps = {
  countries: IMovieDetails['production_countries']
}

export type CountriesHookProps = CountriesProps

export type CountriesHookReturn = {
  formattedCountries: string
  pluralizedTitle: string
}
