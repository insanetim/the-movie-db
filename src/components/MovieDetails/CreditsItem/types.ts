export type CreditsItemHookProps = {
  id: number
  title: string
}

export type CreditsItemHookReturn = {
  handleNavigateToPerson: () => void
}

export type CreditsItemProps = {
  description?: string
  id: number
  profilePath?: string
  title: string
}
