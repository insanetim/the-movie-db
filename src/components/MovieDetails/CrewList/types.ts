import { IMovieCredit } from 'src/interfaces/movie.interface'

export type CrewListProps = {
  crew: IMovieCredit[]
}

export type CrewListHookProps = {
  crew: IMovieCredit[]
}

export type CrewListHookReturn = {
  groupedCrew: {
    groupNames: string[]
  } & Partial<Record<string, IMovieCredit[]>>
}
