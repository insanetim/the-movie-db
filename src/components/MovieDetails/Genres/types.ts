import { IMovieDetails } from 'src/interfaces/movie.interface'

export type GenresProps = {
  genres: IMovieDetails['genres']
}
