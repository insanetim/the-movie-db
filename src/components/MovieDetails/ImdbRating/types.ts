export type ImdbRatingProps = {
  rating: number
}
