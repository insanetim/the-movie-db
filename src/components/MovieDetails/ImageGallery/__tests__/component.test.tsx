import { mockImage } from 'src/__mocks__/mockMovie'
import { renderWithWrapper } from 'src/utils/testHelpers/renderWithWrapper'

import ImageGallery from '../component'
import { ImageGalleryProps } from '../types'

describe('ImageGallery component', () => {
  it('should match snapshot', () => {
    const props: ImageGalleryProps = {
      images: [mockImage],
      title: 'test/title',
    }

    const { asFragment } = renderWithWrapper(<ImageGallery {...props} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
