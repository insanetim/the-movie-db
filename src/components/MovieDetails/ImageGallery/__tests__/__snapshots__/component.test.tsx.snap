// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ImageGallery component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-row css-dev-only-do-not-override-1cj4ms"
  >
    <div
      class="ant-col ant-col-24 css-dev-only-do-not-override-1cj4ms"
    >
      <div
        class="ant-carousel css-dev-only-do-not-override-1cj4ms"
      >
        <div
          class="slick-slider slick-initialized"
          dir="ltr"
        >
          <div
            class="slick-list"
          >
            <div
              class="slick-track"
              style="opacity: 1;"
            >
              <div
                aria-hidden="false"
                class="slick-slide slick-active slick-current"
                data-index="0"
                style="outline: none; width: 0px; position: relative; left: 0px; opacity: 1; z-index: 999; transition: opacity 700ms ease, visibility 700ms ease;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="movie-image"
                    style="width: 100%; display: inline-block;"
                    tabindex="-1"
                  >
                    <img
                      alt="test/title"
                      src="https://image.tmdb.org/t/p/original/image"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <ul
            class="slick-dots slick-dots-bottom"
            style="display: block;"
          >
            <li
              class="slick-active"
            >
              <button>
                1
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
